phpBB/phpbb/console/command/user/add.php

```diff
@@ -0,0 +1,311 @@
+			)
+			->addOption(
+				'password',
+				null,
+				InputOption::VALUE_REQUIRED,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_PASSWORD')
+			)
+			->addOption(
+				'email',
+				null,
+				InputOption::VALUE_REQUIRED,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_EMAIL')
+			)
+			->addOption(
+				'send-email',
+				null,
+				InputOption::VALUE_NONE,
+				$this->language->lang('CLI_DESCRIPTION_USER_ADD_OPTION_NOTIFY')
+			)
+		;
+	}
+
+	/**
+	 * Executes the command user:add
+	 *
+	 * Adds a new user to the database. If options are not provided, it will ask for the username, password and email.
+	 * User is added to the registered user group. Language and timezone default to $config settings.
+	 *
+	 * @param InputInterface  $input  The input stream used to get the options
+	 * @param OutputInterface $output The output stream, used to print messages
+	 *
+	 * @return int 0 if all is well, 1 if any errors occurred
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$io = new SymfonyStyle($input, $output);
+
+		$helper = $this->getHelper('question');
+
+		$data = array(
+			'username'     => $input->getOption('username'),
+			'new_password' => $input->getOption('password'),
+			'email'        => $input->getOption('email'),
+		);
+
+		if (!$data['username'])
+		{
+			$question = new Question($this->ask_user('USERNAME'));
+			$data['username'] = $helper->ask($input, $output, $question);
+		}
+
+		if (!$data['new_password'])
+		{
+			$question = new Question($this->ask_user('PASSWORD'));
+			$question->setValidator(function ($value) use ($helper, $input, $output) {
+				$question = new Question($this->ask_user('CONFIRM_PASSWORD'));
+				$question->setHidden(true);
+				if ($helper->ask($input, $output, $question) != $value)
+				{
+					throw new runtime_exception($this->language->lang('NEW_PASSWORD_ERROR'));
+				}
+				return $value;
+			});
+			$question->setHidden(true);
+			$question->setMaxAttempts(5);
+
+			$data['new_password'] = $helper->ask($input, $output, $question);
+		}
+
+		if (!$data['email'])
+		{
+			$question = new Question($this->ask_user('EMAIL_ADDRESS'));
+			$data['email'] = $helper->ask($input, $output, $question);
+		}
+
+		try
+		{
+			$this->validate_user_data($data);
+			$group_id = $this->get_group_id();
+		}
+		catch (runtime_exception $e)
+		{
+			$io->error($e->getMessage());
+			return 1;
+		}
+
+		$user_row = array(
+			'username'      => $data['username'],
+			'user_password' => $this->password_manager->hash($data['new_password']),
+			'user_email'    => $data['email'],
+			'group_id'      => $group_id,
+			'user_timezone' => $this->config['board_timezone'],
+			'user_lang'     => $this->config['default_lang'],
+			'user_type'     => USER_NORMAL,
+			'user_regdate'  => time(),
+		);
+
+		$user_id = (int) user_add($user_row);
+
+		if ($input->getOption('send-email') && $this->config['email_enable'])
```

Comment:
Shouldn't you also check if user_add() actually returns a user id? Just to be safe that there was no error while adding the user to the database.